libs/membership/api/src/lib/subscription-event-dictionary/subscription-event-dictionary.ts

```diff
@@ -0,0 +1,262 @@
+      throw new Error('Default user subscription flow not found!')
+    }
+
+    if (defaultFlows.length > 1) {
+      throw new Error('Multiple default memberplans found! This is a data integrity error!')
+    }
+
+    const nonDefaultFlows = this.allFlows.filter(flow => !flow.default)
+    if (nonDefaultFlows.filter(flow => !flow.memberPlanId).length > 0) {
+      throw new Error(
+        'Subscription Flow with no memberplan found that is not default! This is a data integrity error!'
+      )
+    }
+
+    this.storeIsBuilt = true
+  }
+
+  /**
+   * Get the earliest date when an invoice must be created.
+   * @param date The current date.
+   * @returns The earliest date.
+   */
+  public getEarliestInvoiceCreationDate(date: Date) {
+    if (!this.storeIsBuilt) {
+      throw new Error('Tried to access store before it was successfully initialized!')
+    }
+    const allIntervals = this.allFlows.map(flow => flow.intervals).flat()
+    const allCreationEvents = allIntervals.filter(
+      interval => interval.event === SubscriptionEvent.INVOICE_CREATION
+    )
+
+    if (allCreationEvents.length === 0) {
+      throw new Error('No invoice creation date found!')
+    }
+
+    let earliest = Number.MAX_SAFE_INTEGER
+    for (const event of allCreationEvents) {
+      if (event.daysAwayFromEnding !== null && event.daysAwayFromEnding < earliest) {
+        earliest = event.daysAwayFromEnding
+      }
+    }
+    return subDays(this.normalizeDate(date), earliest)
+  }
+
+  /**
+   * Get an array of the dates where custom events have been defined.
+   * @param date The current date.
+   * @returns An array of dates.
+   */
+  public getDatesWithCustomEvent(date: Date): Date[] {
+    if (!this.storeIsBuilt) {
+      throw new Error('Tried to access store before it was successfully initialized!')
+    }
+    const customEventDaysAwayFromEnding: number[] = []
+    const allIntervals = this.allFlows.map(flow => flow.intervals).flat()
+    const allCustomEvents = allIntervals.filter(
+      interval => interval.event === SubscriptionEvent.CUSTOM
+    )
+    for (const event of allCustomEvents) {
+      if (
+        event.daysAwayFromEnding !== null &&
+        !customEventDaysAwayFromEnding.includes(event.daysAwayFromEnding)
+      ) {
+        customEventDaysAwayFromEnding.push(event.daysAwayFromEnding)
+      }
+    }
+    return customEventDaysAwayFromEnding.map(daysAwayFromEnding =>
+      subDays(this.normalizeDate(date), daysAwayFromEnding)
+    )
+  }
+
+  /**
+   * Try to get the MailTemplates for a specific filter of [memberPlan,
+   * paymentMethod, periodicity, autoRenewal]. You additionally need to pass
+   * either `daysAwayFromEnding` or `events` to filter further. If any
+   * filter leads to an empty result set, the templates of the default flow are
+   * returned.
+   * @param query The filter of the above properties.
+   * @returns An array of MailTemplates.
+   */
+  public getActionFromStore(query: LookupActionInput): Action[] {
```

Comment:
Question: why method is called getActionFromStore if according to description it gets MailTemplates?
Also plural vs singular.